includes/class-img-converter.php

```
@@ -344,30 +360,63 @@ private function is_animated_webp( $file ) {
+				// Ensure it's still within the WP directory or WP_CONTENT_DIR for safety.
+				$norm_abspath = wp_normalize_path( ABSPATH );
+				$norm_content = wp_normalize_path( WP_CONTENT_DIR );
+				if ( strpos( $local_path, $norm_abspath ) !== 0 && strpos( $local_path, $norm_content ) !== 0 ) {
+					return $source_image;
+				}
+			}
+		}
+
+		// Replace extension.
+		$info       = pathinfo( $local_path );
+		$local_path = wp_normalize_path( $info['dirname'] . '/' . $info['filename'] . '.' . $format );
```

PHPStan (2.1.46)

[warning] Offset 'dirname' might not exist on array{dirname?: string, basename: string, extension?: string, filename: string}.


(offsetAccess.notFound)